CMakeLists.txt

```
@@ -36,27 +36,36 @@ option(GPU_DEVICE
 if("${GPU_DEVICE}" STREQUAL "OFF")
   set(GPU_DEVICE NONE)
 endif()
-if("${GPU_DEVICE}" STREQUAL "NONE")
-  set(GPU_DEVICE_CODE ${GPU_DEVICE_NONE})
-elseif("${GPU_DEVICE}" STREQUAL "CUDA")
-  set(GPU_DEVICE_CODE ${GPU_DEVICE_CUDA})
-elseif("${GPU_DEVICE}" STREQUAL "HIP")
-  # As stated in the PyTorch documentation
-  # (https://docs.pytorch.org/docs/stable/notes/hip.html)
-  # > "PyTorch for HIP intentionally reuses the existing torch.cuda interfaces.
-  # > This helps to accelerate the porting of existing PyTorch code and models
-  # > because very few code changes are necessary, if any."
-  # Therefore we use the CUDA backend when HIP is selected. This is technically
-  # equivalent to specifying -DGPU_DEVICE=CUDA.
-  set(GPU_DEVICE_CODE ${GPU_DEVICE_HIP})
-elseif("${GPU_DEVICE}" STREQUAL "XPU")
-  set(GPU_DEVICE_CODE ${GPU_DEVICE_XPU})
-elseif("${GPU_DEVICE}" STREQUAL "MPS")
-  set(GPU_DEVICE_CODE ${GPU_DEVICE_MPS})
+
+if(NOT "${GPU_DEVICE}" STREQUAL "NONE")
```

I'd make this:
```suggestion
if("${GPU_DEVICE}" STREQUAL "NONE")
  message(STATUS "CPU-only build")
  set(GPU_DEVICE_CODE ${GPU_DEVICE_NONE})
elseif("${GPU_DEVICE}" STREQUAL "CUDA")
  message(STATUS "Building with CUDA support")
  ...
```

which reads clearer IMO and avoids an unnecessary  negation.